### DISCUSSION TITLE
Apple Lossless (ALAC) Support

### DISCUSSION THREAD

**Discussion by petard** [MoleMan1024/audiowagon#146] [Ideas]
As a user, I want to be able to play Apple Lossless music in AudioWagon, so I don't have to maintain and store a music library in FLAC for AudioWagon and one in ALAC for Apple Music.

**Comment by MoleMan1024** [comment]
While Apple has made this codec publicly available here https://github.com/macosforge/alac I will not implement it, it would be too much work and I don't use any Apple devices.

Does Apple Music only support a single file format? (would not be surprising with Apple).

**Comment by petard** [comment]
There is an[ open source Java decoder](https://github.com/soiaf/Java-Apple-Lossless-decoder) for Apple Lossless available, but not sure if that helps.

AFAIK ALAC is the only lossless decoder supported by Apple Music. This makes it a pain to maintain a music library that works on the Music app and in the car via AudioWagon.
